syntax = "proto3";

package store;


service BookStore {
  rpc GetBookRequest(GetReq) returns (BookList) {}
  rpc CreateBookRequest(BookCreateInfo) returns (BookInfo) {}
  rpc EditBookRequest(BookInfo) returns (BookInfo) {}
  rpc DeleteBookRequest(DeleteReq) returns (StatusResponse) {}
}

message GetReq{
  int32 book_id = 1;
  bool get_all = 2;
}

message BookInfo{
  int32 book_id=1;
  string book_title = 2;
  string author = 3;
}

message BookCreateInfo{
  string book_title = 2;
  string author = 3;
}

message  BookList{
  repeated BookInfo book_list = 1;
}

message DeleteReq {
  int32 book_id = 1;
}

message StatusResponse{
  string status = 1;
}

